File: src/main/java/org/example/chessplatformbe/controller/dto/request/CreateArticleDTO.java
package org.example.chessplatformbe.controller.dto.request;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class CreateArticleDTO {
    private Integer updateId;

    @NotBlank(message = "Article title should not be blank")
    private String articleTitle;

    private String imageUrl;

    @NotNull(message = "AuthorId is required")
    private Integer authorId;

    private String contentText;

    private List<Integer> commentsIds;
}

File: src/controller/dto/request/create_article_dto.rs
pub struct CreateArticleDTO {
    update_id: u32,
    article_title: String,
    image_url: String,
    author_id: u32,
    content_text: String,
    comments_ids: Vec<u32>,
}

impl CreateArticleDTO {
    pub fn new(
        update_id: u32,
        article_title: String,
        image_url: String,
        author_id: u32,
        content_text: String,
        comments_ids: Vec<u32>,
    ) -> Self {
        Self {
            update_id,
            article_title,
            image_url,
            author_id,
            content_text,
            comments_ids,
        }
    }
}
